src/supertux/game_object_manager.cpp

```diff
@@ -182,6 +188,15 @@ GameObjectManager::draw(DrawingContext& context)
 void
 GameObjectManager::flush_game_objects()
 {
+  if(s_is_loading)
+    return;
+
+  if(!s_needs_flushing_gameobjects_added &&
+     !s_needs_flushing_gameobjects_removed &&
+     !s_needs_flushing_tilemaps)
+    return;
+
+  if(s_needs_flushing_gameobjects_removed)
   { // cleanup marked objects
     m_gameobjects.erase(
       std::remove_if(m_gameobjects.begin(), m_gameobjects.end(),
```

Comment:
Are we sure not looping to look for objects which have been requested for removal every frame saves time?